`test_binascii` has duplicated tests

# Bug report


```console
$ python -m pip install ruff
...
$ ruff --select F811 Lib/test/test_binascii.py
Lib/test/test_binascii.py:279:9: F811 Redefinition of unused `test_hex_roundtrip` from line 236
Found 1 error.
```

Source: 

https://github.com/python/cpython/blob/f63d37877ad166041489a968233b57540f8456e8/Lib/test/test_binascii.py#L232-L239

https://github.com/python/cpython/blob/f63d37877ad166041489a968233b57540f8456e8/Lib/test/test_binascii.py#L278-L282

The first `test_hex_roundtrip` is doing a `hexlify`/`unhexlify` roundtrip and is correctly named.

The second `test_hex_roundtrip` is doing a `b2a_uu`/`a2b_uu` roundtrip, so can be renamed like `test_b2a_roundtrip`.

This was introduced after https://github.com/python/core-workflow/issues/505#issuecomment-1609495331.

<!-- gh-linked-prs -->
### Linked PRs
* gh-109141
<!-- /gh-linked-prs -->

--- a/Lib/test/test_binascii.py
+++ b/Lib/test/test_binascii.py
@@ -233,7 +233,7 @@ class BinASCIITest(unittest.TestCase):
         binary=hypothesis.strategies.binary(),
         backtick=hypothesis.strategies.booleans(),
     )
-    def test_hex_roundtrip(self, binary, backtick):
+    def test_b2a_roundtrip(self, binary, backtick):
         converted = binascii.b2a_uu(self.type2test(binary), backtick=backtick)
         restored = binascii.a2b_uu(self.type2test(converted))
         self.assertConversion(binary, converted, restored, backtick=backtick)

